.github/workflows/publish_codespace_image.yml

```diff
@@ -0,0 +1,71 @@
+---
+name: Publish ZenML Codespace Docker image
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version tag (leave empty for latest)'
+        required: false
+        type: string
+        default: ''
+  push:
+    tags:
+      - '*'
+    branches:
+      - main
+      - develop
```

Comment:
I don't think we need this action to trigger automatically anywhere. It should build the image at release, and if anything there should be another option to trigger it manually I think